### PR TITLE
Adds feature to generate ICS file to add to calendar

### PR DESCRIPTION
**Pull request by stagrim**
Adds button to create an ICS file using a self created iCalendar library, written in Scala.js. Also adds a checkbox to display times with or without "akademisk kvart", which in turn will carry over to the generated ICS file. I have tested  generated ICS files, but bugs may still occur. 

Please rearrange or style the new UI elements if you have a better idea of how they can be displayed. The same goes for the code itself, if you find a better way of doing things, please contribute.

### CODE REVIEW

**Review by theolundqvist — COMMENTED** [review]
This looks good! Great work creating your own? ICal builder in Scala.

File: src/main/scala/main.scala [hunk 1]
```diff
@@ -2,6 +2,10 @@ package kaptenallocweb
 
 import org.scalajs.dom
 import org.scalajs.dom.document
+import kaptenallocweb.ics.*
+import org.scalajs.dom.Blob
+import org.scalajs.dom.BlobPropertyBag
```
Comment (theolundqvist) [inline src/main/scala/main.scala:7]: If these are needed we could use the direct path instead. 

org.scalajs.dom is already imported.

**Comment by stagrim** [comment]
> This looks good! Great work creating your own? ICal builder in Scala.

The iCalendar library is built from the ground up. It also contains code specifically made for kapten-alloc-web, like the functions in the Property companion object used to create Property classes from kapten-alloc data. More features from the iCalendar standard will be added when they are needed. Currently only features that are needed are implemented. I plan, for example, on adding support for alerts [like my Rust version](https://gitlab.com/stagrim/kapten-alloc-ics) (plan was to use Rust-generated WebAssembly in kapten-alloc-web, but it would have made project more complex to maintain). In that case a new class, Alert, which inherits from VObject would be created, and a VObject vector in Event etc. It is build to be (hopefully) scalable, like to easily add new Parameters, Properties and such. 

https://github.com/bjornregnell/kapten-alloc-web/pull/4#discussion_r753944572 Should be resolved with the 31c8b2d commit

Thanks for the review!

File: src/main/scala/main.scala (hunk 1 above)
Comment (stagrim) [reply src/main/scala/main.scala]: Should be resolved, see https://github.com/bjornregnell/kapten-alloc-web/pull/4#issuecomment-975295391

**Comment by bjornregnell** [comment]
This seems really useful @stagrim ! Happy coding :smile:  
I will take a look hopefully by the end of this week after my lectures if not before.

@theolundqvist thanks for the review! Would be great if you meanwhile also could try to build it locally and perhaps do some quick testing and check the user experience?

@DagHemberg You also expressed interest in this previously. If you have some time over, you are welcome to give it a review if you like.

File: src/main/scala/ics/Property.scala [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+  /** 
+   * Gives formatted Description Property
+   * 
+   * @param course type of course being held
+   * @param group group to be tutored
+   * @param room room held in; will be formatted different is room equals to "Ambulans"
+  */
+  def description(course: String, group: String, room: String): Description =
+    Description(s"Kurs: $course\\nGrupp: $group\\n${ if room.equals("Ambulans") then "Roll" else "Rum" }: $room")
```
Comment (andrephilipsson) [inline src/main/scala/ics/Property.scala:74]: Should probably use `room.startsWith("Ambulans")` to support more "Ambulans" rooms (e.g. "Ambulans1")

File: src/main/scala/ics/Property.scala [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+  /**
+   * @param course type of course being held
+   * @param kind type of session, like "Resurstid" or "Labb"
+   * @param room room held in; will be formatted different is room equals to "Ambulans"
+  */
+  def summary(course: String, kind: String, room: String): Summary =
+    Summary(s"($course) $kind ${ if room.equals("Ambulans") then "som" else "i" } $room")
```
Comment (andrephilipsson) [inline src/main/scala/ics/Property.scala:82]: Same here

**Review by andrephilipsson — COMMENTED** [review]
Works like a charm!

File: src/main/scala/ics/Property.scala (hunk 2 above)
Comment (stagrim) [reply src/main/scala/ics/Property.scala]: Seems like I only fixed that in the Rust version, my bad. Thanks for pointing it out! Fixed in 9e78877.

File: src/main/scala/ics/Property.scala [hunk 4]
```diff
@@ -0,0 +1,109 @@
+package kaptenallocweb.ics
+
+import scala.scalajs.js.defined
+import scala.util.Random
+import java.time.*
+
+/** Give value to a property as if directly given in an ics file */
+trait Property(val value: String) extends BaseObject:
```
[the remark this answers is not in the document]
Comment (bjornregnell) [reply src/main/scala/ics/Property.scala:8]: But this is more nice to have. As we have other reviews with thumbs up I'll merge this for now and perhaps add the above as an issue.

**Review by bjornregnell — APPROVED** [review]

**Comment by bjornregnell** [comment]
I get this build error after merge:
```
sbt:hello Scala.js> fullLinkJS
[info] Full optimizing /home/bjornr/git/hub/bjornregnell/kapten-alloc-web/target/scala-3.1.0/hello-scala-js-opt
[error] file:/home/runner/work/scala-java-time/scala-java-time/core/js/target/scala-3.0.0/src_managed/main/org/threeten/bp/format/DateTimeFormatterBuilder.scala(1536:4:Labeled): any expected but <notype> found for tree of type org.scalajs.ir.Trees$Block
[error] There were 1 IR checking errors.
[error] (Compile / fullLinkJS) There were 1 IR checking errors.
```
Seems to be a clash between Scala versions (?)

**Pull request merged**